src/core/ObservableQuery.ts

```diff
@@ -872,7 +878,9 @@ Did you mean to call refetch(variables) instead of refetch({ variables })?`);
     variables: TVariables | undefined,
   ) {
     const lastError = this.getLastError();
-    if (lastError || this.isDifferentFromLastResult(result)) {
+    if (lastError
+      || this.isDifferentFromLastResult(result)
+      || this.hasDifferentVariablesFromLastResult(variables)) {
```

Comment:
`...FromLastIteration` actually seems like a better name here. Thoughts?

```suggestion
      || this.hasDifferentVariablesFromLastIteration(variables)) {
```